pss/pss.go

```diff
@@ -219,21 +287,29 @@ func (p *Pss) Start(srv *p2p.Server) error {
 			}
 		}
 	}()
-	go func() {
-		for {
-			select {
-			case msg := <-p.outbox:
-				metrics.GetOrRegisterGauge("pss.outbox.len", nil).Update(int64(len(p.outbox)))
 
-				err := p.forward(msg.msg)
-				if err != nil {
-					log.Error(err.Error())
-					metrics.GetOrRegisterCounter("pss.forward.err", nil).Inc(1)
+	// In any case, if the message
+	// forwarding fails, the node should try to forward it to the next best peer, until the message is
+	// successfully forwarded to at least one peer.
+	go func() {
```

Comment:
lets extract that as a function on outbox and start in the initialiser of the outbox no?
the forward function could also be just passed to the outbox constructor, 
an indication is a lot of `p.outbox`-es in the body

Reply:
Sounds good to me.

Reply:
FIxed in 1b91b66